### PR TITLE
Doc update for complex numbers

### PR DESCRIPTION
**Pull request by anjali411**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51129 Doc update for complex numbers**

Differential Revision: [D26094947](https://our.internmc.facebook.com/intern/diff/D26094947)

### CODE REVIEW

File: docs/source/complex_numbers.rst [hunk 1]
```diff
@@ -15,8 +15,8 @@ than operations on float tensors mimicking them. Operations involving complex nu
 to use vectorized assembly instructions and specialized kernels (e.g. LAPACK, cuBlas).
 
 .. note::
-     Spectral operations (e.g., :func:`torch.fft`, :func:`torch.stft` etc.) currently don't use complex tensors but
-     the API will be soon updated to use complex tensors.
+     Spectral operations in the `torch.fft module <https://pytorch.org/docs/stable/fft.html#torch-fft>`_ supports
```
Comment (mruberry) [inline docs/source/complex_numbers.rst:18]: supports -> support

File: docs/source/complex_numbers.rst [hunk 2]
```diff
@@ -107,12 +107,8 @@ The angle and absolute values of a complex tensor can be computed using :func:`t
 Linear Algebra
 --------------
 
-Currently, there is very minimal linear algebra operation support for complex tensors.
-We currently support :func:`torch.mv`, :func:`torch.svd`, :func:`torch.qr`, and :func:`torch.inverse`
-(the latter three are only supported on CPU). However we are working to add support for more
-functions soon: :func:`torch.matmul`, :func:`torch.solve`, :func:`torch.eig`,
-:func:`torch.symeig`. If any of these would help your use case, please
-`search <https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+complex>`_
+We support many linear algebra operations for complex numbers like :func:`torch.matmul`, :func:`torch.svd`, :func:`torch.solve` etc.
```
Comment (mruberry) [inline docs/source/complex_numbers.rst:110]: Suggestion:

"Many linear algebra operations, like ..., support complex numbers."

**Review by mruberry — APPROVED** [review]
Cool!

File: docs/source/complex_numbers.rst [hunk 3]
```diff
@@ -15,8 +15,8 @@ than operations on float tensors mimicking them. Operations involving complex nu
 to use vectorized assembly instructions and specialized kernels (e.g. LAPACK, cuBlas).
 
 .. note::
-     Spectral operations (e.g., :func:`torch.fft`, :func:`torch.stft` etc.) currently don't use complex tensors but
-     the API will be soon updated to use complex tensors.
+     Spectral operations in the `torch.fft module <https://pytorch.org/docs/stable/fft.html#torch-fft>`_ support
+     native complex tensors.
```
Comment (albanD) [inline docs/source/complex_numbers.rst:19]: nit: "complex tensors natively"?

File: docs/source/complex_numbers.rst [hunk 4]
```diff
@@ -107,12 +107,8 @@ The angle and absolute values of a complex tensor can be computed using :func:`t
 Linear Algebra
 --------------
 
-Currently, there is very minimal linear algebra operation support for complex tensors.
-We currently support :func:`torch.mv`, :func:`torch.svd`, :func:`torch.qr`, and :func:`torch.inverse`
-(the latter three are only supported on CPU). However we are working to add support for more
-functions soon: :func:`torch.matmul`, :func:`torch.solve`, :func:`torch.eig`,
-:func:`torch.symeig`. If any of these would help your use case, please
-`search <https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+complex>`_
+Many linear algebra operations, like :func:`torch.matmul`, :func:`torch.svd`, :func:`torch.solve` etc., support complex numbers.
```
Comment (albanD) [inline docs/source/complex_numbers.rst:110]: Should the `,` be before the `etc.`?

File: docs/source/complex_numbers.rst [hunk 5]
```diff
@@ -131,12 +127,12 @@ Complex tensors can be serialized, allowing data to be saved as complex values.
 Autograd
 --------
 
-PyTorch supports autograd for complex tensors. The autograd APIs can be
-used for both holomorphic and non-holomorphic functions. For holomorphic functions,
-you get the regular complex gradient. For :math:`C → R` real-valued loss functions,
-`grad.conj()` gives a descent direction. For more details, check out the note :ref:`complex_autograd-doc`.
+PyTorch supports autograd for complex tensors. The gradient computed is the Conjugate Wirtinger derivative,
+the negative of which is precisely the direction of steepest descent used in Gradient Descent algorithm. Thus,
```
Comment (albanD) [inline docs/source/complex_numbers.rst:131]: nit: that should be used in Gradient Descent

**Review by albanD — APPROVED** [review]
lgtm just some nits

**Pull request closed without merging**